### PR TITLE
Merge release 1.55.2 to develop

### PR DESCRIPTION
**Pull request by dcalhoun**
Merges latest 1.55.2 release into develop branch. 

To test: Verify the `gutenberg` ref points to the [merge commit](https://github.com/WordPress/gutenberg/pull/32895#event-4929858602) of the release branch. 

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -4,6 +4,10 @@ Unreleased
 * [*] Tablet view fixes for inserter button. https://github.com/wordpress-mobile/gutenberg-mobile/pull/3602
 * [**] Fixed an issue where pressing enter inside a text-based block was not creating a new block when using Gboard [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3590]
 
+1.55.2
+------
+* [**] Fix incorrect block insertion point after blurring the post title field. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3640]
```
Comment (fluiddot) [inline RELEASE-NOTES.txt:8]: I think we have this entry duplicated in the `Unreleased` section so we could remove it from there.

**Comment by fluiddot** [comment]
I noticed that the `Check Correctness ` CI job is failing due to differences in the `package-lock.json` file, maybe we need to run `npm install` and update the file, wdyt?

**Comment by dcalhoun** [comment]
> I noticed that the `Check Correctness ` CI job is failing due to differences in the `package-lock.json` file, maybe we need to run `npm install` and update the file, wdyt?

I had done so once before after the initial `Check Correctness` failure, so I am puzzled as to why it is still failing. I'll try `npm install` again in hopes of getting passing results.

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (dcalhoun) [reply RELEASE-NOTES.txt:8]: Good catch. Addressed in https://github.com/wordpress-mobile/gutenberg-mobile/pull/3655/commits/5ba6364e562714cf9d23e242be69ce243dc0bc2b.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**Pull request merged**